BuildScript/BuildTasks/MetadataValueDictionary.cs

```diff
@@ -32,17 +32,11 @@ public MetadataValueDictionary (IDictionary<string, string> rawDictionary)
       _rawDictionary = new Dictionary<string, string> (rawDictionary, StringComparer.OrdinalIgnoreCase);
     }
 
-    public string GetValue (IEnumerable<string> keys, Func<Exception> createEmptySequenceException)
+    public KeyValuePair<string, string> GetKeyValuePair (IEnumerable<string> possibleKeys, Func<Exception> createEmptySequenceException)
     {
-      return keys
-                 .Select (x => _rawDictionary.GetValueOrDefault (x))
-                 .SingleOrDefault (x => x != null)
-             ?? throw createEmptySequenceException();
-    }
+      var key = _rawDictionary.Keys.Intersect (possibleKeys, StringComparer.OrdinalIgnoreCase).SingleOrDefault() ?? throw createEmptySequenceException();
```

Comment:
That Single here will throw if there's multiple matches using a useless excpetion message. In order to identify which set of keys returned multiple matches, we need to build the check without relying on Single